doc/lstm.txt

```diff
@@ -1,14 +1,177 @@
 .. _lstm:
 
-LSTM Network for Sentiment Analysis
+LSTM Networks for Sentiment Analysis
 **********************************************
 
 Summary
 +++++++
 
-This tutorial aims to provide an example of how a Recurrent Neural Network (RNN) using the Long Short Term Memory (LSTM) architecture can be implemented using Theano. In this tutorial, this model is used to perform sentiment analysis on movie reviews from the `Large Movie Review Dataset <http://ai.stanford.edu/~amaas/data/sentiment/>`_, sometimes known as the IMDB dataset. 
+This tutorial aims to provide an example of how a Recurrent Neural Network
+(RNN) using the Long Short Term Memory (LSTM) architecture can be implemented
+using Theano. In this tutorial, this model is used to perform sentiment
+analysis on movie reviews from the `Large Movie Review Dataset
+<http://ai.stanford.edu/~amaas/data/sentiment/>`_, sometimes known as the
+IMDB dataset.
+
+In this task, given a movie review, the model attempts to predict whether it
+is positive or negative. This is a binary classification task.
+
+Data
+++++
+
+As previously mentionned, the provided scripts are used to train a LSTM
+recurrent neural on the Large Movie Review Dataset dataset.
+
+While the dataset is public, in this tutorial we provide a copy of the dataset
+that has previously been preprocessed according to the needs of this LSTM
+implementation. You can download this preprocessed version of the dataset
+using the script `download.sh
+<https://raw.githubusercontent.com/lisa-lab/DeepLearningTutorials/master/data/download.sh>`_
+and uncompress it.
+
+Model
++++++
+
+LSTM
+====
+
+In a *traditional* recurrent neural network, during the gradient
+back-propagation phase, the gradient signal can end up being multiplied a
+large number of times (as many as the number of timesteps) by the weight
+matrix associated with the connections between the neurons of the recurrent
+hidden layer. This means that, the magnitude of weights in the transition
+matrix can have a strong impact on the learning process.
+
+If the weights in this matrix are small, it can lead to a situation called
+*vanishing gradients* where the gradient signal gets so small that learning
+either becomes very slow or stops working altogether. It can also make more
+difficult the task of learning long-term dependencies in the data.
+Conversely, if the weights in this matrix are large, it can lead to a
+situation where the gradient signal is so large that it can cause learning to
+diverge. This is often referred to as *exploding gradients*.
+
+These issues are the main motivation behind the LSTM model which introduces a
+new structure called a *memory cell* (see Figure 1 below). A memory cell is
+composed of four main elements: an input gate, a neuron with a self-recurrent
```

Comment:
I do not think so. However, you can have a "memory cell block" (see http://web.eecs.utk.edu/~itamar/courses/ECE-692/Bobby_paper1.pdf page 12 paragraph 4.4). This is a group of memory cells that share both input gave and output gate.